### PR TITLE
add GET /alert/newNotificationsCount endpoint (CU-hjwcwk)

### PR DESCRIPTION
**Pull request by schwarrrtz**
- the Alert App uses this to display the number of new notifications
- also added the notifications table to the db
- also updated and added tests

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-hjwcwk Frontend; Home Screen: Implementation of notifications](https://app.clickup.com/t/hjwcwk)

**Comment by schwarrrtz** [comment]
I tested this by running the existing test suite along with a new test I wrote that covers the new code. I also tested this by deploying it on `chatbot-dev.brave.coop` and connecting the new frontend code (see https://github.com/bravetechnologycoop/brave-alert-app/pull/16). everything seems to be working as expected.

File: server/db/015-addnotifications.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    -- Get the migration ID of the last file to be successfully run
+    SELECT MAX(id) INTO lastSuccessfulMigrationId
+    FROM migrations;
+
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+        CREATE TABLE IF NOT EXISTS notifications (
+            id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+            installation_id uuid REFERENCES installations (id),
```
Comment (trisapeace) [inline server/db/015-addnotifications.sql:16]: I think that `installation_id` needs an explicit `NOT NULL` (https://stackoverflow.com/questions/28206232/can-foreign-key-references-contain-null-values-in-postgresql)

File: server/test/integration/BraveAlerterConfiguratorTest/getAlertSessionByPhoneNumberTest.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ describe('BraveAlerterConfigurator.js integration tests: getAlertSessionByPhoneN
     this.installationIncidentCategories = ['Cat1', 'Cat2', 'Cat3']
 
     await db.clearSessions()
+    await db.clearNotifications()
```
Comment (trisapeace) [inline server/test/integration/BraveAlerterConfiguratorTest/getAlertSessionByPhoneNumberTest.js:21]: For all the tests where this was added in the `beforeEach`, could you please also add it in the `afterEach`?

File: server/test/integration/BraveAlerterConfiguratorTest/getNewNotificationsCountByAlertApiKeyTest.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    await db.createNotification(this.installationId, 'subject', 'body', false)
+    await db.createNotification(this.installationId, 'subject', 'body', false)
+    await db.createNotification(this.installationId, 'subject', 'body', true)
+
+    const braveAlerterConfigurator = new BraveAlerterConfigurator()
+    this.braveAlerter = braveAlerterConfigurator.createBraveAlerter()
+  })
+
+  afterEach(async () => {
+    await db.clearNotifications()
```
Comment (trisapeace) [inline server/test/integration/BraveAlerterConfiguratorTest/getNewNotificationsCountByAlertApiKeyTest.js:31]: Could you please add `clearSessions()` in the `afterEach` as well?

**Review by trisapeace — COMMENTED** [review]
Looks good. I just pointed out a couple of small things.

File: server/db/015-addnotifications.sql (hunk 1 above)
Comment (schwarrrtz) [reply server/db/015-addnotifications.sql:16]: gotcha, good catch.

**Review by trisapeace — APPROVED** [review]
Sweet. Thanks!

**Review by mariocimet — APPROVED** [review]
looks good and satsified w the testing done

**Pull request merged**